templates/flavors/base/cluster-template.yaml

```diff
@@ -69,6 +67,7 @@ spec:
         extraArgs:
           cloud-provider: azure
           cloud-config: /etc/kubernetes/azure.json
+          feature-gates: "WindowsHostProcessContainers=true"
```

Comment:
This addition to the base `Cluster` flavor definition (and the resultant additive explosion to lots of cluster scenario templates) is to enable a single calico configuration that works whether or not you have Windows nodes in your cluster. For example:

```
Error: INSTALLATION FAILED: DaemonSet.apps "kube-proxy-windows" is invalid: spec.template.spec.securityContext.windowsOptions.hostProcess: Forbidden: not allowed when feature gate 'WindowsHostProcessContainers' is not enabled
helm.go:84: [debug] DaemonSet.apps "kube-proxy-windows" is invalid: spec.template.spec.securityContext.windowsOptions.hostProcess: Forbidden: not allowed when feature gate 'WindowsHostProcessContainers' is not enabled
INSTALLATION FAILED
```

More broadly, turning on this feature gate by default enables Windows nodes to be added to a cluster with little friction after the cluster has been created. We should be leaning into "Windows is a possible cluster configuration that we always enable for all clusters" rather than "Windows is an intrinsic feature that you either enable or disable at cluster creation time".

cc @jsturtevant @marosset @CecileRobertMichon